plugins/woocommerce/src/Admin/Features/Blueprint/Init.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+declare( strict_types = 1);
+
+namespace Automattic\WooCommerce\Admin\Features\Blueprint;
+
+use Automattic\WooCommerce\Admin\PageController;
+
+/**
+ * Class Init
+ *
+ * This class initializes the Blueprint feature for WooCommerce.
+ */
+class Init {
+	/**
+	 * Init constructor.
+	 */
+	public function __construct() {
+		add_action( 'rest_api_init', array( $this, 'init_rest_api' ) );
+		add_filter( 'woocommerce_admin_shared_settings', array( $this, 'add_upload_nonce_to_settings' ) );
+	}
+
+	/**
+	 * Register REST API routes.
+	 *
+	 * @return void
+	 */
+	public function init_rest_api() {
+		( new RestApi() )->register_routes();
+	}
+
+
+	/**
+	 * Add upload nonce to global JS settings.
+	 *
+	 * The value can be accessed at wcSettings.admin.blueprint_upload_nonce
+	 *
+	 * @param array $settings
+	 *
+	 * @return array
+	 */
+	public function add_upload_nonce_to_settings( array $settings ) {
```

Comment:
Should we check if the feature flag is enabled before adding the nonce?

Reply:
Same comment as [above](https://github.com/woocommerce/woocommerce/pull/49765#discussion_r1692009934).